### ISSUE TITLE
Write a mapper to truncate to N chars

### ISSUE THREAD

**Issue by jimduff-usds** [CDCgov/prime-data-hub#670]
From customer at Arizona about Recovery in Motion:

> We have an issue with what is being populated in the Lab_Name field for this facility. Currently what is being populated is "Recovery in Motion Treatment Center, Recovery in Motion Treatment Center". Our ELR database can only accept a max value of 50 characters for that component in accordance with the HL7 standards. This value is over 50 characters. Please resolve.

It seems like the most straightforward way is a TruncateMapper(filename, maxlength)

Or, you could do someone one-off because this field happens to be a concatenation, eg, if its over 50 chars, just display the first part.

**Comment by jimduff-usds** [comment]
@MauriceReeves-usds , what are your thoughts on this? 

Here's what I'm thinking:

We should allow combinations of mappers, but it looks like that's not a feature right now.   If we had that, we could write a general truncate(field, len) mapper and run it after the concatenate mapper.

Without that, the low-hanging-fruit solution is simply to add a final maxLength field to the concatenate(...) mapper.

`concatenate("field1_name" "field2_name", ... "fieldNName", 50)`

**Comment by MauriceReeves-usds** [comment]
This is something I've also discussed with @RickHawesUSDS because there are a few fields that other receivers want truncated (in their case to 20 chars). In their case it's related to HL7 compliance, so we discussed setting a flag on translation that would then trigger a truncation wrapper for certain fields, or something in the translation config:

```
mappers:
    - truncate(50, field_name)
    - truncate(20, field_name, field_name2, field_name3)
```

Adding a max length to the `concatenate` mapper is the easiest route to solve this for AZ.  I would probably start there, but I'm already adding suppression logic to the translation config for OH, so we might end up moving everything there.

**Comment by jimduff-usds** [comment]
@MauriceReeves-usds, how about this idea:  add a new maxLength attribute to TEXT and similar fields.   This should be fairly easy to implement, and also much more general than adding a maxLength to just one mapper.
?

**Comment by RickHawesUSDS** [comment]
I'm wondering if limit is should be an Element member. It such a universal attribute. All HL7 elements have a specification limit.

**Comment by jimduff-usds** [comment]
Yes, I've implemented a new option Element field called maxLength.  Its only operational for certain text fields, and only then if it appears in the schema with a valid value.

Example, that solves AZ's request:

```
  # Nonstandard
  - name: organization_and_facility
    type: TEXT
    maxLength: 50
    mapper: 'concat(organization_name, ordering_facility_name)'
    csvFields: [{name: Ordering_facility_name}]

  # Nonstandard
  - name: organization_and_lab
    type: TEXT
    maxLength: 50
    mapper: 'concat(organization_name, testing_lab_name)'
    csvFields:
    - name: Lab_Name
```

**Comment by jimduff-usds** [comment]
Fixed in PR #694
